porch/engine/pkg/engine/options.go

```diff
@@ -69,6 +70,21 @@ func WithGRPCFunctionRuntime(address string) EngineOption {
 	})
 }
 
+func WithKubeFunctionRuntime(cl client.Client, ns string) EngineOption {
+	return EngineOptionFunc(func(engine *cadEngine) error {
+		runtime, err := newKubeFunctionRuntime(cl, ns)
+		if err != nil {
+			return fmt.Errorf("failed to create function runtime: %w", err)
+		}
+		if engine.runtime == nil {
+			engine.runtime = runtime
+		} else {
+			engine.runtime = fn.NewMultiRuntime([]fn.FunctionRuntime{engine.runtime, runtime})
```

Comment:
same problem as in the other PR where we end up with tree of runtimes instead of a list.